swarm/network/protocol.go

```diff
@@ -214,10 +217,25 @@ func (b *Bzz) performHandshake(p *protocols.Peer, handshake *HandshakeMsg) error
 		return err
 	}
 	handshake.peerAddr = rsh.(*HandshakeMsg).Addr
+	sanitizeEnodeRemote(p.RemoteAddr(), handshake.peerAddr)
 	handshake.LightNode = rsh.(*HandshakeMsg).LightNode
 	return nil
 }
 
+// the remote enode string may advertise arbitrary host information (e.g. localhost)
+// this method ensures that the addr of the peer will be the one
+// applicable on the interface the connection came in on
+// it modifies the passed bzzaddr in place, and returns the same pointer
+func sanitizeEnodeRemote(paddr net.Addr, baddr *BzzAddr) {
+	hsSubmatch := regexpEnodeIP.FindSubmatch(baddr.UAddr)
+	ip, _, err := net.SplitHostPort(paddr.String())
+	if err == nil && string(hsSubmatch[1]) != ip {
```

Comment:
up until this point you have no guaranty that `hsSubmatch` worked correctly. You might be getting an out of bound error when accessing `hsSubmatch[1]`

Reply:
Technically you are correct, yes.

However, we _know_ that the `UAddr` is the enode string, _and_ we know exactly how that enode string looks. Since the input is predictable, I don't think it's necessary to double check it.

Reply:
you are assuming IPv4 structure. What happens if an IPv6 address finds its way as a node address?

Reply:
We are using ipv4 explicitly, so it won't until we refactor the code, and by then this code will be gone anyway :)

Reply:
If you feel strongly about it, I can put it in, though <3

Reply:
If it is guarantied that all addresses would be v4 then this is a non-issue.